modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1403 @@
+  return data;
+}
+
+pragma "no doc"
+private extern proc sys_sendto(sockfd:fd_t, buff:c_void_ptr, len:c_long, flags:c_int, const ref address:sys_sockaddr_t,  ref num_sent_out:ssize_t):err_t;
+
+/*
+  Send `data` over socket to the provided address and
+  return number of bytes sent if successful.
+
+  .. code-block:: Chapel
+
+    const timeout = new timeval(4,0);
+    const socket = new udpSocket();
+    const sentBytes = socket.send("hello world!":bytes, timeout);
+
+  :arg data: data to send to address
+  :type data: :mod:`bytes <Bytes>`
+  :arg address: socket address for sending data
+  :type address: :type:`ipAddr`
+  :arg timeout: time to wait for data to arrive.
+  :type timeval: :type:`~Sys.timeval`
+  :return: sentBytes
+  :rtype: `ssize_t`
+  :throws SystemError: Upon failure to send any data
+                        within given `timeout`.
+*/
+proc udpSocket.send(data: bytes, in address: ipAddr, in timeout = new timeval(-1,0)):ssize_t throws {
+  var err_out:err_t = 0;
+  var length:ssize_t;
+  err_out = sys_sendto(this.socketFd, data.c_str():c_void_ptr, data.size:c_long, 0, address._addressStorage, length);
+  if err_out == 0 {
+    return length;
+  }
+  if err_out != 0 && err_out != EAGAIN && err_out != EWOULDBLOCK {
+    throw SystemError.fromSyserr(err_out, "send failed");
+  }
+  var localSync$: sync c_short;
+  var internalEvent = event_new(event_loop_base, this.socketFd, EV_WRITE | EV_TIMEOUT, c_ptrTo(syncRWTCallback), c_ptrTo(localSync$):c_void_ptr);
+  defer {
+    event_free(internalEvent);
+  }
+  while err_out != 0 {
+    var t: Timer;
+    t.start();
+    if timeout.tv_sec == -1 {
+      err_out = event_add(internalEvent, nil);
+    }
+    else {
+      err_out = event_add(internalEvent, c_ptrTo(timeout));
+    }
+    if err_out != 0 {
+      throw SystemError.fromSyserr(err_out, "send failed");
+    }
+    var retval = localSync$.readFE();
+    t.stop();
+    if retval & EV_TIMEOUT != 0 {
+      throw SystemError.fromSyserr(ETIMEDOUT, "send timed out");
+    }
+    var elapsedTime = t.elapsed(TimeUnits.microseconds):c_long;
+    err_out = sys_sendto(this.socketFd, data.c_str():c_void_ptr, data.size:c_long, 0, address._addressStorage, length);
+    if err_out != 0 {
+      if err_out != EAGAIN && err_out != EWOULDBLOCK {
+        throw SystemError.fromSyserr(err_out, "send failed");
+      }
+      if timeout.tv_sec == -1 {
+        var totalTimeout = timeout.tv_sec*1000000 + timeout.tv_usec;
+        if totalTimeout >= t.elapsed(TimeUnits.microseconds) {
+          const remainingMicroSeconds = ((totalTimeout - elapsedTime)%1000000);
+          const remainingSeconds = ((totalTimeout - elapsedTime)/1000000);
+          timeout.tv_sec = remainingSeconds;
+          timeout.tv_usec = remainingMicroSeconds;
+        }
+        throw SystemError.fromSyserr(ETIMEDOUT, "send timed out");
+      }
+    }
+  }
+  return length;
+}
+
+inline operator !=(const ref lhs: udpSocket,const ref rhs: udpSocket) {
```

Comment:
Similar, `no doc` or document and/or mention the support for comparison in a header comment.